python/mlx/nn/losses.py

```diff
@@ -372,3 +372,45 @@ def log_cosh_loss(
     loss = mx.logaddexp(errors, -errors) - math.log(2)
 
     return _reduce(loss, reduction)
+
+
+def cosine_similarity_loss(
+    inputs: mx.array,
+    targets: mx.array,
+    eps: float = 1e-8,
+    reduction: str = "none",
```

Comment:
Could you make the `axis` a parameter to this function instead of hardcoding it?